[pt-br] Add tasks/run-application/scale-stateful-set.md (#51257)
kubernetes/website: content/pt-br/docs/tasks/run-application/scale-stateful-set.md

```shell
kubectl scale statefulsets <stateful-set-name> --replicas=<new-replicas>
```

### Faça atualizações in-place nos seus StatefulSets

Alternativamente, você pode fazer
[atualizações in-place](/docs/concepts/cluster-administration/manage-deployment/#in-place-updates-of-resources)
em seus StatefulSets.

Se o seu StatefulSet foi criado inicialmente com `kubectl apply`,
atualize o `.spec.replicas` dos manifestos do StatefulSet e, em seguida, execute um `kubectl apply`:

```shell
kubectl apply -f <stateful-set-file-updated>
```

Caso contrário, edite esse campo com `kubectl edit`:

```shell
kubectl edit statefulsets <stateful-set-name>
```

Ou use `kubectl patch`:

```shell
kubectl patch statefulsets <stateful-set-name> -p '{"spec":{"replicas":<new-replicas>}}'
```

## Solução de problemas

### Reduzir o escalonamento não funciona corretamente

Você não pode reduzir o escalonamento de um StatefulSet enquanto qualquer um dos Pods
com estado que ele gerencia estiver com problemas de saúde. A redução do escalonamento
só ocorre depois que esses Pods com estado estiverem em execução e prontos.

Se `spec.replicas` > 1, o Kubernetes não consegue determinar o motivo de um Pod com estado

> estar com problemas de saúde. Isso pode ser resultado de uma falha permanente ou de uma falha transitória.

Reviewer: O mesmo que o parágrafo anterior aqui:
```suggestion
não estar íntegro. Isso pode ser resultado de uma falha permanente ou de uma falha transitória.
```

Uma falha transitória pode ser causada por uma reinicialização necessária devido a uma atualização ou manutenção.

> Se o Pod estiver com problemas de saúde devido a uma falha permanente, escalonar sem corrigir

Reviewer: ```suggestion
Se o Pod não estiver íntegro devido a uma falha permanente, escalonar sem corrigir
```

a falha pode levar a um estado em que a quantidade de membros do StatefulSet fique abaixo
do número mínimo de réplicas necessário para funcionar corretamente.
Isso pode fazer com que seu StatefulSet se torne indisponível.

> Se o Pod estiver com problemas de saúde devido a uma falha transitória e o Pod possa voltar a ficar disponível,
> o erro transitório pode interferir na sua operação de aumento ou redução de escalonamento.
> Alguns bancos de dados distribuídos apresentam problemas quando nós entram e saem ao mesmo tempo. Nesses casos,
> é melhor analisar as operações de escalonamento no nível da aplicação e realizar o escalonamento apenas quando
> você tiver certeza de que o cluster da sua aplicação com estado está completamente saudável.

Reviewer: ```suggestion
Se o Pod não estiver íntegro devido a uma falha transitória e o Pod possa voltar a ficar disponível,
o erro transitório pode interferir na sua operação de aumento ou redução de escalonamento.
Alguns bancos de dados distribuídos apresentam problemas quando nós entram e saem ao mesmo tempo. Nesses casos,
é melhor analisar as operações de escalonamento no nível da aplicação e realizar o escalonamento apenas quando
você tiver certeza de que o cluster da sua aplicação com estado está completamente íntegro.
```
